### PR TITLE
[doc] Move the c-extensions documentation to no-member doc

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

When the original documentation was created the full list of message did not exists, so this piece of documentation was always a little out of place.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "messages/messages_introduction": "../user_guide/messages/index.html",
     "messages/messages_list": "../user_guide/messages/messages_overview.html",
     "user_guide/message-control": "messages/message_control.html",
-    "technical_reference/c_extensions": "../user_guide/c_extensions.html",
+    "technical_reference/c_extensions": "../user_guide/messages/error/no-member.html",
```
Comment (Pierre-Sassoulas) [inline doc/conf.py:64]: We changed that not so long ago, I think it's find to not maintain the intermediate URL.

File: doc/data/messages/n/no-member/details.rst [hunk 2]
```diff
@@ -1,9 +1,9 @@
-Pylint and C extensions
-=======================
-
 If you are getting the dreaded **no-member** error, there is a possibility that
-either **pylint** found a bug in your code or that it actually tries to lint
-a C extension module.
+either:
```
Comment (DanielNoord) [inline doc/data/messages/n/no-member/details.rst:2]: Let's remove the weird `bold` ``**pylint**`` or use it consistently.

**Review by DanielNoord — APPROVED** [review]
One little comment!

File: doc/data/messages/n/no-member/details.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+- pylint found a bug in your code
+- You're launching pylint without the dependencies installed in its environment.
+- pylint would need to lint a C extension module and is refraining to do so.
+
+Linting C extension modules is not supported out of the box, especially since
+pylint has no way to get an AST object out of the extension module.
+
+But pylint actually has a mechanism which you might use in case you
+want to analyze C extensions. pylint has a flag, called ``extension-pkg-allow-list``
```
Comment (DanielNoord) [inline doc/data/messages/n/no-member/details.rst:12]:
```suggestion
want to analyze C extensions. Pylint has a flag, called ``extension-pkg-allow-list``
```

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**